pubsub/awspubsub/awspubsub.go

```diff
@@ -135,41 +136,25 @@ var stringDataType = aws.String("String")
 
 // SendBatch implements driver.Topic.SendBatch.
 func (t *topic) SendBatch(ctx context.Context, dms []*driver.Message) error {
-	// The result for each message (either nil or an error) will be written to
-	// errc and collected at the end. We'll return the first non-nil error
-	// (if any).
-	errc := make(chan error, len(dms))
+	g, ctx := errgroup.WithContext(ctx)
 	var ctxErr error
 	for _, dm := range dms {
-		// If ctx is done, don't start any more goroutines. We can't just return
-		// because we want to make sure we wait until any goroutines we've already
-		// started complete.
-		if ctxErr != nil {
-			errc <- ctxErr
-			continue
-		}
 		select {
 		case <-ctx.Done():
 			ctxErr = ctx.Err()
-			errc <- ctxErr
+			break
```

Comment:
This is a no-op here. It just breaks you out of the select. https://golang.org/ref/spec#Break_statements. You'll need to label the loop.

Reply:
Ooh, good catch. Done.